### DISCUSSION TITLE
Forming IEEE 802.15.4 Frames with Raw Link

### DISCUSSION THREAD

**Discussion by orangenlimer159** [openthread/openthread#9805] [Q&A] [answered]
Hi OpenThread Community,

This might be a dumb question..., but I've been experimenting with the transmission of MAC frames using the `otLinkRawTransmit()` function from the Raw Link API. However, I'm having difficulty in forming the IEEE 802.15.4 frame. Could anyone provide examples or guidance on how to do this properly? 

Thanks in advance!

**Comment by jwhui** [comment, marked as answer]
You can find some examples in `mac_frame.cpp`, which provides a number of methods for generating different types of MAC frames.

For example, take a look at:
https://github.com/openthread/openthread/blob/c73829fdbf5b0019dc611e913674021492c3fa9e/src/core/mac/mac_frame.cpp#L1402
